services/user-management/src/main/java/com/foodie/user/controllers/RolePermissionController.java

```diff
@@ -0,0 +1,183 @@
+package com.foodie.user.controllers;
+
+import com.foodie.FoodieBaseResponse;
+import com.foodie.user.contracts.PermissionRequest;
+import com.foodie.user.contracts.RolePermitRequest;
+import com.foodie.user.model.Permission;
+import com.foodie.user.model.Role;
+import com.foodie.user.model.RolePermission;
+import com.foodie.user.service.PermissionService;
+import com.foodie.user.service.RolePermissionService;
+import com.foodie.user.service.RoleService;
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@Slf4j
+@RestController
+@CrossOrigin
+@RequestMapping("/api/v1/security/{role}/permissions")
+public class RolePermissionController {
+
+    private final RoleService roleService;
+    private final PermissionService permissionService;
+    private final RolePermissionService rolePermissionService;
+
+    public RolePermissionController(RoleService roleService, PermissionService permissionService,
+            RolePermissionService rolePermissionService) {
+        this.roleService = roleService;
+        this.permissionService = permissionService;
+        this.rolePermissionService = rolePermissionService;
+    }
+
+    private Role validateAndGetRole(UUID roleId) {
```

Comment:
Consider moving these out of your controller. This is against Single responsibility rule